iosHyperskillApp/iosHyperskillApp/Sources/Modules/StepQuiz/Views/StepQuizView.swift

```diff
@@ -126,13 +126,16 @@ struct StepQuizView: View {
                     StepQuizStatsView(text: formattedStats)
                 }
 
-                if viewModel.stepRoute is StepRouteLearn {
-                    ProblemsLimitAssembly(
-                        appearance: .init(showTopDivider: true)
-                    ).makeModule()
+                if case .content(let problemsLimitContentState) = viewModel.problemsLimitViewStateKs,
+                   case .widget = ProblemsLimitFeatureViewStateContentKs(problemsLimitContentState) {
+                    Divider()
                 }
 
-                buildQuizStatusView(state: state, attemptLoadedState: attemptLoadedState)
+                ProblemsLimitContent(
+                    stateKs: viewModel.problemsLimitViewStateKs
+                )
```

Comment:
You can create concrete problems limit view (`StepQuizProblemsLimitView`) for step quiz that will encapsulate this logic with a divider